### PR TITLE
[2.2.x] Graceful Shutdown Refactoring #1154

### PR DESCRIPTION
**Pull request by ryanemerson**
No longer explicitly delete pods on GracefulShutdown, instead scale the StatefulSet once all of the caches have been shutdown on the server.

### CODE REVIEW

**Review by rigazilla — APPROVED** [review]
lgtm

File: pkg/infinispan/cluster.go [hunk 1]
```diff
@@ -127,18 +127,23 @@ func (c Cluster) GracefulShutdownTask(podName string) error {
...
+	   stdErr.println("Encountered error trying to determine cache shutdown order: " + err);
+	}
 
-	var internalCaches = new HashSet(icr.getInternalCacheNames());
-	/* The ___script_cache is included in both getCacheNames() and getInternalCacheNames so prevent repeated shutdown calls */
-	internalCaches.removeAll(cacheNames);
-	shutdown(internalCaches);
+	cachesToShutdown.addAll(icr.getInternalCacheNames());
+	cachesToShutdown.addAll(cacheManager.getCacheNames());
+	shutdown(cachesToShutdown);
```
Comment (danberindei) [inline pkg/infinispan/cluster.go:146]: Shouldn't we define the shutdown function before invoking it?

**Comment by Crumby** [comment]
@ryanemerson Would it make sense to update gracefulShutdown test to use 2 nodes? The test itself would uncover the issue if executed on OpenShift 4.9+

**Comment by ryanemerson** [comment]
> Would it make sense to update gracefulShutdown test to use 2 nodes? The test itself would uncover the issue if executed on OpenShift 4.9+

I've increased the number of nodes to 2, but the test won't fail unless we add a write to one of the caches that encounters [the exception](https://issues.redhat.com/browse/ISPN-13121?focusedCommentId=19442375&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-19442375). Obviously I can't add that test to the CI until we have a new server image with the fix, so I think I'll create a follow up PR with this change

File: config/manager/manager.yaml [hunk 2]
```diff
@@ -37,7 +37,7 @@ spec:
               fieldRef:
                 fieldPath: metadata.namespace
           - name: RELATED_IMAGE_OPENJDK
-            value: "quay.io/infinispan/server:13.0.1.Final"
+            value: "quay.io/infinispan/server:13.0.2.Final"
```
Comment (Crumby) [inline config/manager/manager.yaml:40]: Shouldn't this be 13.0.3?

Comment (ryanemerson) [reply config/manager/manager.yaml:40]: :+1: It wasn't available when I created the PR and was required for CI. I'll be changing this to 13.0.3.Final as part of the release commit.

**Pull request merged**